src/__tests__/span_processor.test.ts

```diff
@@ -0,0 +1,157 @@
+import { Client } from "../client"
+import { SpanProcessor } from "../span_processor"
+import { Tracer, BasicTracerProvider } from "@opentelemetry/sdk-trace-base"
+import { trace, context } from "@opentelemetry/api"
+import { Span } from "../span"
+
+describe("Span processor", () => {
+  let tracer: Tracer
+  let tracerProvider: BasicTracerProvider
+  let client: Client
+
+  function mockClient() {
+    const originalCreateOpenTelemetrySpan =
+      client.extension.createOpenTelemetrySpan
+
+    jest
+      .spyOn(client.extension, "createOpenTelemetrySpan")
+      .mockImplementation((...args) => {
+        const span = originalCreateOpenTelemetrySpan(...args)
+        mockSpan(span)
+        return span
+      })
+  }
+
+  function mockSpan(span: Span) {
```

Comment:
What did you learn writing this about how this solution is better? Did the span registry you referenced have gaps anywhere?

What I like about the span registry is that we're able to load it into other test files as well. In case we ever want to assert span values in any other test files.

Reply:
The advantage of replacing the class implementation (and by replacing we mean extending it with some additional testing behavior), is that we don't need to spy on a specific instance of the AppSignal client/extension. It works without spies! What you say is true, spans are only ever created by the `createOpenTelemetrySpan` function on the extension, but it's also never created without a class instance of `Span`. Which I why opted for the Span class replacement solution.

As [I said in my commit back then](https://github.com/appsignal/appsignal-nodejs/commit/86b42af3704a9ded22365e706ba65667a3ea3600), I was inspired by what I wrote for the Ruby gem, but I also see that even that is not a perfect implementation and a test mode in the extension would be even better. Find a way for `spanToJson` to return the attributes even when the span is closed rather than hack our way around it.
I remember seeing a spec in the OpenTelemetry package that they assert what's being sent through a (HTTP) exporter, which I don't know if we could do reliably in every test setup.

Reply:
I'll grab the SpanTestRegistry from the main branch and use it on these tests, modifying it to work with the new Span class. I liked the original implementation and it should be easy to adapt it for this use case.